howso/engine/session.py

```diff
@@ -288,6 +275,7 @@ def from_dict(cls, session_dict: dict) -> "Session":
         for key in cls.attribute_map.keys():
             if key in session_dict:
                 setattr(instance, f'_{key}', session_dict[key])
+        instance.client = client
```

Comment:
Ah looks like this also accepts the client in the dict. Lets have the method param take precendence over this. It should also be passed in during class init, rather than set afterwards.
i.e. `'client': client or session_dict.get('client')`

Reply:
Addressed by: 2d4a23a